src/Gzero/Core/Query/Condition.php

```diff
@@ -181,7 +187,8 @@ protected function validate()
         if (!in_array($this->operation, self::$allowedOperations, true)) {
             throw new InvalidArgumentException('Unsupported condition operation');
         }
-        if (is_array($this->value) && !$this->isCorrectRangeFormat()) {
+        if (is_array($this->value) && !$this->isCorrectRangeFormat()
+            && !$this->operation === 'in' && !$this->operation === 'not in') {
```

Comment:
I don't like this part, those kinds of conditions are in `isCorrectRangeFormat` method, I think that we need a separate exception which checks if this is array operation and value is an array.

If the operation is for an array and value is not an array it will throw the exception.
Ten we should check `isCorrectRangeFormat` which will throw the exception for between operations.